### PR TITLE
[SCREENREADER]: Education STEM Update - Step 6 Edit button needs an updated aria-label #762

### PR DESCRIPTION
**Pull request by zurbergram**
## Description
The changes in this PR are in response to https://github.com/department-of-veterans-affairs/va.gov-team/issues/762

The issue is 
1) Chapter name != page name
2) When there's only 1 page in the chapter, the page name is hidden
3) The edit button edits the page contents, and says the page name when using SCREENREADER or JAWS, which wasn't visible and therefore is the cause of some confusion

The proposed fix is to show the page title if it doesn't match the chapter title. 

## Testing done
- local testing

### How to test
The goal of these steps is the have a chapter with only one page and the chapter title does not equal the page title on the review screen.  The 1995 form chapter Military History in the STEM flow matches this criteria.

1. Go to http://localhost:3001/education/apply-for-education-benefits/application/1995/introduction
2. Login or start without signing in
3. Proceed to http://localhost:3001/education/apply-for-education-benefits/application/1995/benefits/stem
4. Select "Yes"
    - then "yes" or "no" then "yes"
    - See first or second screenshot below
5. Proceed to review page of application
6. Notice that "Applicant Information" chapter **does not** have a page title of "Applicant information" next to the "Edit" button for its single page (see third screenshot below)
7. Notice that "Military History" chapters **does** have a page title of "Active Duty" next to the "Edit" button for its single page (see third screenshot below)

You can confirm chapter titles and page titles by either viewing at https://github.com/department-of-veterans-affairs/vets-website/blob/master/src/applications/edu-benefits/1995/config/chapters.js

## Screenshots
Screen Shot 2019-08-06 at 9 14 49 AM
Screen Shot 2019-08-06 at 9 14 57 AM
Screen Shot 2019-08-06 at 9 15 25 AM

## Acceptance criteria
- [ ] Page titles are shown when there is a single page in a chapter and the page title does not equal the chapter title

## Definition of done
- [ ] Events are logged appropriately
- [ ] Documentation has been updated, if applicable
- [ ] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [ ] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

**Review by shawkey-daniel — APPROVED** [review]
LGTM

**Review by cvalarida — COMMENTED** [review]
I like it! We'll get some designer eyes on it hopefully tomorrow to make sure it passes muster, but I talked with @1Copenut today and he confirms it makes a lot of sense. :+1: 

Can you make sure this new functionality has a test written for it?

**Review by sonntag-adam — APPROVED** [review]

**Comment by 1Copenut** [comment]
@desireeturner && @cvalarida 
I discussed this PR and the design implications with @smoonpeyes this morning. She agrees the decision made here is correct and a design improvement.

In testing it locally, I noticed a couple of page headings that were repeated. I opened a new issue https://github.com/department-of-veterans-affairs/va.gov-team/issues/830 to capture this and tagged our content specialist in the description.

**Comment by zurbergram** [comment]
> I like it! We'll get some designer eyes on it hopefully tomorrow to make sure it passes muster, but I talked with @1Copenut today and he confirms it makes a lot of sense. 👍
> 
> Can you make sure this new functionality has a test written for it?

Added unit tests

**Comment by desireeturner** [comment]
@1Copenut Thank you. For department-of-veterans-affairs/va.gov-team#830, should we wait for guidance before from @peggygannon before we begin resolution or can we go ahead and get started? Thank you!

File: src/platform/forms-system/src/js/review/ReviewCollapsibleChapter.jsx [hunk 1]
```diff
@@ -67,6 +68,16 @@ export default class ReviewCollapsibleChapter extends React.Component {
     );
   }
 
+  shouldHideExpandedPageTitle = (expandedPages, chapterTitle, pageTitle) => {
+    if (environment.isProduction()) {
```
Comment (cvalarida) [inline src/platform/forms-system/src/js/review/ReviewCollapsibleChapter.jsx]: Why do we need this check?

File: src/platform/forms-system/test/js/review/ReviewCollapsibleChapter.unit.spec.jsx [hunk 2]
```diff
@@ -375,6 +377,113 @@ describe('<ReviewCollapsibleChapter>', () => {
...
+    const pages = [
+      {
+        pageKey: 'test',
+        title: testPageTitle,
+        updateFormData: (oldData, newData) => ({ ...newData, bar: 'baz' }),
+      },
+    ];
+    const chapterKey = 'test';
+    const chapter = {
+      title: 'test chapter title',
```
Comment (cvalarida) [inline src/platform/forms-system/test/js/review/ReviewCollapsibleChapter.unit.spec.jsx]: Could use the `testChapterTitle` variable here too.

File: src/platform/forms-system/test/js/review/ReviewCollapsibleChapter.unit.spec.jsx [hunk 3]
```diff
@@ -375,6 +377,113 @@ describe('<ReviewCollapsibleChapter>', () => {
...
+        onEdit={onEdit}
+        open
+        expandedPages={pages}
+        chapterKey={chapterKey}
+        chapterFormConfig={chapter}
+        form={form}
+      />,
+    );
+
+    expect(wrapper.exists('form-review-panel-page-header')).to.be.false;
```
Comment (cvalarida) [inline src/platform/forms-system/test/js/review/ReviewCollapsibleChapter.unit.spec.jsx]: What was happening when you checked for a zero-length?

```js
const titleDiv = wrapper.find('.form-review-panel-page-header');

expect(titleDiv.length).to.equal(0);
```

Comment (zurbergram) [reply src/platform/forms-system/test/js/review/ReviewCollapsibleChapter.unit.spec.jsx]: AssertionError: expected 1 to equal 0

but 

    wrapper.find('.form-review-panel-page-header').text()

prints out "target Element not found"

File: src/platform/forms-system/test/js/review/ReviewCollapsibleChapter.unit.spec.jsx (hunk 2 above)
Comment (zurbergram) [reply src/platform/forms-system/test/js/review/ReviewCollapsibleChapter.unit.spec.jsx]: fixed

File: src/platform/forms-system/src/js/review/ReviewCollapsibleChapter.jsx (hunk 1 above)
Comment (zurbergram) [reply src/platform/forms-system/src/js/review/ReviewCollapsibleChapter.jsx]: removed

File: src/platform/forms-system/test/js/review/ReviewCollapsibleChapter.unit.spec.jsx (hunk 3 above)
Comment (zurbergram) [reply src/platform/forms-system/test/js/review/ReviewCollapsibleChapter.unit.spec.jsx]: Fixed test

File: src/platform/forms-system/test/js/review/ReviewCollapsibleChapter.unit.spec.jsx [hunk 4]
```diff
@@ -375,6 +377,116 @@ describe('<ReviewCollapsibleChapter>', () => {
...
+        expandedPages={pages}
+        chapterKey={chapterKey}
+        chapterFormConfig={chapter}
+        form={form}
+      />,
+    );
+
+    const titleDiv = wrapper.find('.form-review-panel-page-header');
+    expect(titleDiv.length).to.equal(1);
+    expect(titleDiv.text()).to.equal('');
```
Comment (cvalarida) [inline src/platform/forms-system/test/js/review/ReviewCollapsibleChapter.unit.spec.jsx]: Ahh, that's how it works. :+1:

File: src/platform/forms-system/test/js/review/ReviewCollapsibleChapter.unit.spec.jsx [hunk 5]
```diff
@@ -375,6 +377,116 @@ describe('<ReviewCollapsibleChapter>', () => {
...
+    expect(titleDiv.text()).to.not.equal(testChapterTitle);
```
Comment (cvalarida) [inline src/platform/forms-system/test/js/review/ReviewCollapsibleChapter.unit.spec.jsx]: Probably don't need this here.

**Review by cvalarida — APPROVED** [review]
Looks great! Thanks for this! :raised_hands: :shipit:

File: src/platform/forms-system/test/js/review/ReviewCollapsibleChapter.unit.spec.jsx (hunk 5 above)
Comment (zurbergram) [reply src/platform/forms-system/test/js/review/ReviewCollapsibleChapter.unit.spec.jsx]: removed

**Comment by 1Copenut** [comment]
@desireeturner I'd say to move ahead with the change set, but talk to @peggygannon as soon as possible. Since it's going to surface the same label 3 times (chapter button and 2 page titles), it'd be good to have the right language in place.

**Comment by desireeturner** [comment]
@1Copenut Will do! Thanks

**Pull request merged**